### ISSUE TITLE
RFC: Decoding [msg=] BBcode for mail

### ISSUE THREAD

**Issue by bttrx** [auge8472/My-Little-Forum-1#53]
Hi!

currently posting text `As mentioned in [msg=16101]IMPORTANT: Forum migration in November 2019[/msg] version 1.7.8 lacks full UTF-8 support.` will be rendered to `As mentioned in IMPORTANT: Forum migration in November 2019 --> 16101 version 1.7.8 lacks full UTF-8 support`.

Please notice the `16101`. That should be a clickable link to the posting. So mlf should place the correct URL to the forum in front of as in "Link to the entry: ...". Either take it from mlf's default view or the user's default view.

I have no idea for the `IMPORTANT: Forum migration in November 2019` part. Maybe enclose in " or [] to make it stand out from normal message text?

Robert

**Comment by auge8472** [comment]
This looks like the output, that should be created for e-mails. You see this when you open the posting  in the web browser?

**Comment by bttrx** [comment]
Yes, it's the output for mails. Sorry, I didn't finish my first sentence in the initial posting before clicking "Submit new issue".

So, what I like to say is: "16101" is just plain text. No link to anywhere. It should be something like `https://[...]/forum/forum_entry.php?id=16101` instead.

**Comment by auge8472** [comment]
That's, like it works since ever and yes, the proposed behaviour is the preferable behaviour.

**edit:** to make it more clear, if you add a link to a posting, that contains a complete URL, you get a working URL in the e-mail. When Alex added the bb-code `msg`, the change wasn't adapted to [the function unbbcode](https://github.com/auge8472/My-Little-Forum-1/blob/cc2193f3906171c3ee5d4b25548fd74aa71cfa0b/functions.php#L224), where the posting text get prepared for the output in an e-mail.

**Comment by bttrx** [comment]
> That's, like it works since ever

Yes. I were not clear enough, that this is *not* a *new* bug.

> if you add a link to a posting, that contains a complete URL, you get a working URL in the e-mail

You mean, [msg=url]text[/msg] is also valid? Never tried that before.
